File: updater/src/main/java/com/legatotechnologies/updater/UpdatePreference.java
package com.legatotechnologies.updater;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
/**
 * Created by davidng on 6/21/17.
 */

public class UpdatePreference {
    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;
    private static final String PREF_NOTIFICATION_TIME = "laterNotificationTime";
    private static final String PREF_UPDATE_LATER = "updateLater";
    private static final String PREF_SKIP_VERSION = "skipVersion";
    private static final String PREF_LAST_VERSION = "lastVersion";

    UpdatePreference(Context context) {
        this.sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        this.editor = sharedPreferences.edit();
    }

    boolean isUpdateLater () {
        return sharedPreferences.getBoolean(PREF_UPDATE_LATER, false);
    }

    void setUpdateLater(boolean value) {
        editor.putBoolean(PREF_UPDATE_LATER, value)
                .commit();
    }

    long getLaterNotificationTime() {
        return sharedPreferences.getLong(PREF_NOTIFICATION_TIME, 0L);
    }

    void setLaterNotificationTime(long value) {
        editor.putLong(PREF_NOTIFICATION_TIME, value)
                .commit();
    }

    String getSkipVersion() {
        return sharedPreferences.getString(PREF_SKIP_VERSION, null);
    }

    void setSkipVersion(String value) {
        editor.putString(PREF_SKIP_VERSION, value)
                .commit();
    }

    String getLastVersion(){
        return sharedPreferences.getString(PREF_LAST_VERSION, null);
    }

    void setLastVersion(String version){
        editor.putString(PREF_LAST_VERSION, version)
                .commit();
    }
}

File: updater/src/main/java/com/legatotechnologies/v2/updater/UpdatePreference.kt
package com.legatotechnologies.v2.updater

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import com.legatotechnologies.v2.updater.preference.SharePrefUtil

class UpdatePreference internal constructor(context: Context) {
    private val preferences: SharePrefUtil = SharePrefUtil.with(context)
        .setName("force_update_prefs")
        .setSecure(false)
        .ok()

    var isUpdateLater: Boolean
        get() = preferences.readBoolean(PREF_UPDATE_LATER, false)
        set(value) = preferences.saveBoolean(PREF_UPDATE_LATER, value)

    var laterNotificationTime: Long
        get() = preferences.readLong(PREF_NOTIFICATION_TIME, 0L)
        set(value) = preferences.saveLong(PREF_NOTIFICATION_TIME, value)

    var skipVersion: String
        get() = preferences.readString(PREF_SKIP_VERSION, "")
        set(value) = preferences.saveString(PREF_SKIP_VERSION, value)

    var lastVersion: String
        get() = preferences.readString(PREF_LAST_VERSION, "")
        set(version) = preferences.saveString(PREF_LAST_VERSION, version)

    companion object {
        private const val PREF_NOTIFICATION_TIME = "later_notification_time"
        private const val PREF_UPDATE_LATER = "update_later"
        private const val PREF_SKIP_VERSION = "skip_version"
        private const val PREF_LAST_VERSION = "last_version"
    }
}